public/app/features/alerting/unified/rule-list/FilterView.tsx

```diff
@@ -107,9 +114,25 @@ function FilterViewResults({ filterState }: FilterViewProps) {
   return (
     <Stack direction="column" gap={0}>
       <ul aria-label="filtered-rule-list">
-        {rules.map(({ key, rule, groupIdentifier }) => (
-          <AlertRuleLoader key={key} rule={rule} groupIdentifier={groupIdentifier} />
-        ))}
+        {rules.map((ruleWithOrigin) => {
+          const { key, rule, groupIdentifier, origin } = ruleWithOrigin;
+
+          switch (origin) {
+            case 'grafana':
```

Comment:
A high level comment as well, as I'm looking at this/wrapping my head around the changes - do we want to use constants for `grafana` and `datasource` for the origins? As we reference them in a few places now (`grafana` could be a separate one to the `GRAFANA_RULES_SOURCE` that we use everywhere already)

Reply:
I'm not sure. We could use `GRAFANA_RULES_SOURCE`, on the other hand there is no risk of typos, as typescript won't allow incorrect values